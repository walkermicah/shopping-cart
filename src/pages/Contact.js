import '../styles/pages/Contact.scss';

export default function Contact() {
  return (
    <div className="Contact">
      <form className="Contact-form">
        <h3>Contact Us</h3>

        <label>
          <span>Your name</span>
          <input className="Contact-form-name" type="text" name="name" />
        </label>

        <label>
          <span>Email</span>
          <input className="Contact-form-email" type="email" name="email" />
        </label>

        <label>
          <span>Message</span>
          <textarea className="Contact-form-message" name="message" />
        </label>

        <button type="submit">Submit</button>
      </form>
    </div>
  );
}
