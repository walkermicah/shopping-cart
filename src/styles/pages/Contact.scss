@import '../abstracts/mixins';

.Contact {
  @include background-img;
  padding: 2rem;

  display: flex;
  justify-content: center;

  &-form {
    @include page-content;
    padding: 0rem 3rem;
    width: 45%;
    background-color: black;

    h3 {
      text-align: center;
      padding: 1.5rem;
      color: white;
      font-size: 4rem;
      word-spacing: 1rem;
    }

    label {
      display: grid;
      padding: 0rem 2rem;
    }

    span {
      color: white;
      padding: 0.5rem;
      font-weight: 300;
      border-radius: 3px;
    }

    &-name,
    &-email,
    &-message {
      border: none;
      border-radius: 3px;
      padding: 1rem;
    }

    &-name:focus,
    &-email:focus,
    &-message:focus {
      outline: none;
      border-bottom: 5px solid rgb(193, 43, 43);
    }

    &-message {
      height: 15rem;
      resize: none;
    }

    button {
      justify-self: center;
      padding: 1rem 2rem;
      cursor: pointer;
      color: white;
      background-color: rgb(0, 0, 0, 1);
      border: 1px solid white;
      border-radius: 3px;
      text-transform: uppercase;
      font: inherit;
      font-size: 2.5rem;
      font-weight: 400;
      transition: all 0.3s;
    }
  }

  button:hover {
    transform: translateY(-1px);
  }
}

/* 1000px  */
@media only screen and (max-width: 62.5em) {
  .Contact-form {
    width: 60%;
  }
}

/* 700px  */
@media only screen and (max-width: 43.75em) {
  .Contact-form {
    width: 75%;
  }
}

/* 600px  */
@media only screen and (max-width: 37.5em) {
  .Contact-form {
    width: 95%;
  }
}
