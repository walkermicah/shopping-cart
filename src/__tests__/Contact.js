import { render } from '@testing-library/react';
import Contact from '../pages/Contact';

describe('Contact Page Component', () => {
  it('Renders contact form page', () => {
    const { container } = render(<Contact />);
    expect(container).toMatchSnapshot();
  });
});
