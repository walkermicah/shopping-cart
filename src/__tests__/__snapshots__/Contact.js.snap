// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Page Component Renders contact form page 1`] = `
<div>
  <div
    class="Contact"
  >
    <form
      class="Contact-form"
    >
      <h3>
        Contact Us
      </h3>
      <label>
        <span>
          Your name
        </span>
        <input
          class="Contact-form-name"
          name="name"
          type="text"
        />
      </label>
      <label>
        <span>
          Email
        </span>
        <input
          class="Contact-form-email"
          name="email"
          type="email"
        />
      </label>
      <label>
        <span>
          Message
        </span>
        <textarea
          class="Contact-form-message"
          name="message"
        />
      </label>
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
